airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionNotificationWorkflowImpl.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling;
+
+import io.airbyte.api.client.invoker.generated.ApiException;
+import io.airbyte.commons.temporal.scheduling.ConnectionNotificationWorkflow;
+import io.airbyte.config.Notification;
+import io.airbyte.config.Notification.NotificationType;
+import io.airbyte.config.SlackNotificationConfiguration;
+import io.airbyte.config.StandardSync;
+import io.airbyte.config.persistence.ConfigNotFoundException;
+import io.airbyte.notification.SlackNotificationClient;
+import io.airbyte.validation.json.JsonValidationException;
+import io.airbyte.workers.temporal.annotations.TemporalActivityStub;
+import io.airbyte.workers.temporal.scheduling.activities.ConfigFetchActivity;
+import io.airbyte.workers.temporal.scheduling.activities.NotifySchemaChangeActivity;
+import io.airbyte.workers.temporal.scheduling.activities.SlackConfigActivity;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ConnectionNotificationWorkflowImpl implements ConnectionNotificationWorkflow {
+
+  @TemporalActivityStub(activityOptionsBeanName = "shortActivityOptions")
+  private NotifySchemaChangeActivity notifySchemaChangeActivity;
+  @TemporalActivityStub(activityOptionsBeanName = "shortActivityOptions")
+  private SlackConfigActivity slackConfigActivity;
+  @TemporalActivityStub(activityOptionsBeanName = "shortActivityOptions")
+  private ConfigFetchActivity configFetchActivity;
+
+  @Override
+  public boolean sendSchemaChangeNotification(UUID connectionId)
```

Comment:
This might be a style issue, we can discuss this outside of this PR if it helps.

The activity breakdown doesn't feel natural to me which brings up a few questions:
* Do we really want to have one activity per API request? I find that tedious to follow, code end up very truncated, having to hop through interfaces/implementations to read. Side note, probably not a concern here due to the load of notifications, but technically, that's a few extra round trips, in terms of networking.
* Looking at line 41: intuitively, I think this bit of logic should be done in the `notifySchemaChangeActivity` rather than in the workflow.

Reply:
Yes I see what you're saying here. I'll make this a followup issue